plugins/action/netconf.py

```diff
@@ -23,86 +20,18 @@
 
 class ActionModule(ActionNetworkModule):
     def run(self, tmp=None, task_vars=None):
-        del tmp  # tmp no longer has any effect
-
         module_name = self._task.action.split(".")[-1]
         self._config_module = True if module_name == "netconf_config" else False
         persistent_connection = self._play_context.connection.split(".")[-1]
         warnings = []
 
-        if persistent_connection not in ["netconf", "local"] and module_name == "netconf_config":
-            return {
-                "failed": True,
-                "msg": "Connection type %s is not valid for netconf_config module. "
-                "Valid connection type is netconf or local (deprecated)"
-                % self._play_context.connection,
-            }
-        elif persistent_connection not in ["netconf"] and module_name != "netconf_config":
+        if persistent_connection != "netconf":
```

Comment:
Aren't there plugins in junos collection that support network_cli?

Reply:
(side note: we need to remove the note that says local is supported in JunOS modules wherever applicable).

Reply:
There are, but this action plugin is exclusively for the `netconf_*` plugins: `netconf_config`, `netconf_get`, and `netconf_rpc`. Of them, only `netconf_config` supported local, and as that is no longer supported for that module, there is no reason to have the branch.

Reply:
re: junos, https://github.com/ansible-collections/junipernetworks.junos/blob/main/plugins/action/junos.py handles the connection checking for those modules. And is probably doing too much work to support local that isn't limited to the two modules that require `connection: local`: `junos_package` and `junos_scp`. When someone goes back to clean up those docs I might make sure that `if connection == local` check also checks for one of those two modules to be named and strip out any functionality that doesn't concern them.

Reply:
Ah, nvm. I confused this file with the junos action plugin. My bad.